### PR TITLE
Refactor the functions using Redux toolkit and API.

### PR DESCRIPTION
**Pull request by MoRaad97**
### In this PR
- Add an asynchronous reducer with axois to get the books from API.
- Add an asynchronous reducer with axois to add book to the book list.
- Add an asynchronous reducer with axois to removebook to the book list.
- Implement the function to react components.
- Fix Linters errors.

### CODE REVIEW

File: src/Redux/configureStore.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { configureStore } from '@reduxjs/toolkit';
+// import thunk from 'redux-thunk';
```
Comment (youmari) [inline src/Redux/configureStore.js]: - [x] please try to remove the unused commented code to keep the file clean and pro

File: src/Components/addBookForm.js [hunk 2]
```diff
@@ -1,25 +1,53 @@
-import React, { useRef } from 'react';
-import { useDispatch } from 'react-redux/es/exports';
-import uniqid from 'uniqid';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
 import styles from './Styles/addBookForm.module.css';
-import { addBook } from '../Redux/books/books';
+// import { useDispatch } from 'react-redux/es/hooks/useDispatch';
```
Comment (youmari) [inline src/Components/addBookForm.js]: - [x]  please try to remove the unused commented code to keep the file clean and pro

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @MoRaad97 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great title and short summary 
✔️ Githubflow
✔️ No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/addBookForm.js (hunk 2 above)
Comment (MoRaad97) [reply src/Components/addBookForm.js]: Thanks @youmari for the feedback, 
working on changes now 👍.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @MoRaad97 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**